src/avt/Blueprint/avtConduitBlueprintDataAdaptor.C

```diff
@@ -3323,3 +3326,129 @@ avtConduitBlueprintDataAdaptor::BlueprintToMFEM::FieldToMFEM(
+    mfem::Mesh *mesh,
+    const Node &n_field)
+{
+    bool zero_copy = true;
+    // n_conv holds converted data (when necessary for mfem api)
+    // if n_conv is used ( !n_conv.dtype().empty() ) we
+    // know that some data allocation was necessary, so we
+    // can't return a qf that zero copies the conduit data
+    Node n_conv;
+
+    const double *vals_ptr = NULL;
+    int vdim = 1;
+
+    if (n_field["values"].dtype().is_object())
+    {
+        vdim = n_field["values"].number_of_children();
+
+        // need to check that we have doubles and
+        // cover supported layouts
+        if ( n_field["values"][0].dtype().is_double() )
+        {
+            // quad funcs use what mfem calls byVDIM
+            // and what conduit calls interleaved
+            // check for interleaved
+            if (blueprint::mcarray::is_interleaved(n_field["values"]))
+            {
+                // conduit mcarray interleaved == mfem byVDIM
+                vals_ptr = n_field["values"].child(0).value();
+            }
+            else
+            {
+                // for mcarray generic case --  default to byVDIM
+                // aka interleaved
+                blueprint::mcarray::to_interleaved(n_field["values"],
+                                                  n_conv["values"]);
+                vals_ptr = n_conv["values"].child(0).value();
+            }
+        }
+        else // convert to doubles and use interleaved
+        {
+            Node n_tmp;
+            // check all vals, if we don't have doubles convert
+            // to doubles
+            NodeConstIterator itr = n_field["values"].children();
+            while (itr.has_next())
+            {
+                const Node &c_vals = itr.next();
+                std::string c_name = itr.name();
+
+                if ( c_vals.dtype().is_double() )
+                {
+                    // zero copy current coords
+                    n_tmp[c_name].set_external(c_vals);
+
+                }
+                else
+                {
+                    // convert
+                    c_vals.to_double_array(n_tmp[c_name]);
+                }
+            }
+
+            // for mcarray generic case --  default to byVDIM
+            // aka interleaved
+            blueprint::mcarray::to_interleaved(n_tmp,
+                                               n_conv["values"]);
+            vals_ptr = n_conv["values"].child(0).value();
+        }
+    }
+    else // scalar case
+    {
+        if (n_field["values"].dtype().is_double() &&
+            n_field["values"].is_compact())
+        {
+            vals_ptr = n_field["values"].value();
+        }
+        else
+        {
+            n_field["values"].to_double_array(n_conv["values"]);
+            vals_ptr = n_conv["values"].value();
+        }
+    }
+
+    if (zero_copy && !n_conv.dtype().is_empty())
+    {
+        //Info: "Cannot zero-copy since data conversions were necessary"
+        zero_copy = false;
+    }
```

Comment:
zero copy is always true at this point so you don't need to ask if it is, unless you meant to modify it before this point?

Reply:
I think you responded to the wrong comment. I was talking originally about the `zero_copy` variable above, which is always true by the time we get to the if-statement.

Reply:
yes, I see what you mean .